### PR TITLE
update to  go1.11

### PR DESCRIPTION
**Pull request by komuw**
# what
- updated travisCI to go1.11
- updated the script to check dependencies to check for changes in go.mod and go.sum files

# why
- This fixes; https://github.com/gomods/athens/issues/379

This PR is dependent on PR; https://github.com/gomods/athens/pull/371 been approved and merged.

### CODE REVIEW

**Comment by michalpristas** [comment]
also bear in mind that proxy dockerfile refers go1.11rc not rc2 so it will not popup in your search

**Comment by komuw** [comment]
@michalpristas @robjloranger @marwan-at-work 
I have updated references of Go1.11 RCs
I've also created `GO111MODULE=auto` and `GO111MODULE=on` since I think we should  test that athens works in both modes atleast until Go1.12 ships and `GO111MODULE=on` becomes the default

File: cmd/proxy/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.11-rc
+FROM golang:1.11
```
Comment (marwan-at-work) [inline cmd/proxy/Dockerfile]: I don't think `golang:1.11` is out on Dockerhub yet, we can do this in another PR or just make this PR a DO NOT MERGE one until it gets published: https://hub.docker.com/r/library/golang/tags/

Comment (komuw) [reply cmd/proxy/Dockerfile]: @marwan-at-work thanks for catching that. 
I've added WIP to the PR. 
I'll remove that once the image get uploaded. 

The PR[1] to do that was created 30 minutes ago, so I expect it to be merged and uploaded soon.

1. https://github.com/docker-library/official-images/pull/4779

**Comment by robjloranger** [comment]
I just changed the title to reflect all changes this PR will make.

**Comment by komuw** [comment]
@marwan-at-work 
the image has been released on dockerHub
```bash
docker run -it golang:1.11 go version
go version go1.11 linux/amd64
```

**Comment by michalpristas** [comment]
pushing this in, great work @komuw
